### PR TITLE
Some species mutation tweaks/fixes

### LINKED ISSUE #46192: Genetic mutations and becoming an android
Issue reported from  Round ID: 117186 (/tg/Station Bagil [ENGLISH] [US-WEST] [100% FREE LAG])

Reporting client version: 512.1472

I stayed around a delaminating SM without protection for too long and got the Radioactive mutation. Then, I was fed an apple that turned me into an Android. The mutation remained, sadly.
Trying to fix it revealed some problems. Mutadone didn't do anything. Killing and cloning just had me clone in as an android with the same mutation. Someone attempted to use a genetics machine on me, but said he could not manipulate my genes at all. I tried to podperson out, but couldn't draw blood because androids cannot be syringed and I'm not sure if I had blood or not.
Was about to attempt a brain transplant, but didn't get the opportunity.

Basically, I think it's strange that mutations persist in Androids, who don't seem designed to work with genetics, preventing the issue from being fixed, in a way. Sorry if this was rambly or useless.

### PR DESCRIPTION
**Pull request by SteelSlayer**
## About The Pull Request
EDIT: If you become a species who is rad immune, you will lose any and all mutations you have, as rad immune species shouldn't be able to get mutations normally.

Fixes #46192

## Changelog
:cl:
tweak: Upon becoming a species that is rad immune, you will lose any mutations you have
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -271,6 +271,9 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	if(DIGITIGRADE in species_traits)
 		C.Digitigrade_Leg_Swap(FALSE)
 
+	if(TRAIT_RADIMMUNE in inherent_traits)
```
Comment (Couls) [inline code/modules/mob/living/carbon/human/species.dm]: You need to move this down to https://github.com/tgstation/tgstation/blob/9aee2702e7e8d453bfd938aae0d256870fa7673c/code/modules/mob/living/carbon/human/species.dm#L311

**Review by Couls — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/modules/mob/living/carbon/human/species.dm]: Yeah that will look better. I didn't even see those down there..

**Comment by Time-Green** [comment]
You should probably just remove mutations from all species without DNA

**Comment by nemvar** [comment]
If you only remove the radioactive mutation from rad immune species that should probably be handled by the mutation itself. (*cough* signal)

But it probably makes more sense to just remove all mutations since rad immune species can't actually recieve mutations by normal means.

**Comment by SteelSlayer** [comment]
Yeah that makes sense. I just noticed, you can't change peoples DNA in the genetics machine if they have rad immunity.

**Pull request merged**